_posts/2019-08-06-serverless-static-sites.md

```diff
@@ -0,0 +1,232 @@
+---
+
+Learn how you can continuosly deploy static sites with Hugo, OpenFaaS Cloud (or GitLab) and the FunctionIngress CRD.
+
+## Pre-req
+* Kubernetes
+
+You need to have a Kubernetes cluster up & running. I recommend [DigitalOcean](https://www.digitalocean.com/), it's cost-efficient and very easy to set-up.
+
+* [Install Helm](https://github.com/openfaas/faas-netes/blob/master/HELM.md)
+
+You'll need helm to install the components in the cluster. If you can't install `Tiller`, helm's server-side component, in your cluster then you can use `helm template`.
+
+* [Install OpenFaaS using Helm](https://github.com/openfaas/faas-netes/blob/master/chart/openfaas/README.md#install)
+
+For this tutorial you will need to install OpenFaaS using an ingress controller instead of exposing the gateway with a LoadBalancer. For this you'll need to set `ingress.enabled=true` and `exposeServices=false` when installing it.
+
+* Install an Ingress Controller
+
+Currently we support [Nginx](https://github.com/kubernetes/ingress-nginx), [Traefik](https://docs.traefik.io/user-guide/kubernetes/) and Zalando's [Skipper](https://opensource.zalando.com/skipper/kubernetes/ingress-controller/). We will be using Nginx for this tutorial:
+```sh
+helm install stable/nginx-ingress --name nginxingress --set rbac.create=true
+```
+
+* [Install the cert-manager](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#10-ssl-for-the-gateway) (optional)
+
+Installing the cert-manager to get automatic TLS certificates is optional but I highly recommend it. I will be using it today.
+
+## Install the Ingress Operator
+The Ingress Operator automatically manages the creation of custom ingress rules and TLS certificates for your functions using a new CRD called `FunctionIngress` introduced by Alex in the [previous blog post](https://www.openfaas.com/blog/custom-domains-function-ingress/).  
+Lets deploy the operator to our cluster following the instructions at [the documentation](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/#20-ssl-and-custom-domains-for-functions):
+```sh
+git clone https://github.com/openfaas-incubator/ingress-operator
+cd ingress-operator
+
+kubectl apply -f ./artifacts/operator-crd.yaml
+kubectl apply -f ./artifacts/operator-rbac.yaml
+kubectl apply -f ./artifacts/operator-amd64.yaml
+```
+
+## Introducing the hugo template
+Today I'm showing you a new template for creating [hugo](https://gohugo.io) static sites that took me little to no time to make thanks to the simplicity and flexibility of [OpenFaaS's templating system](https://www.openfaas.com/blog/template-store/). You can check it out [here](https://github.com/matipan/openfaas-hugo-template).  
+The template copies over the contents of your hugo site, builds it into the `public` directory and then uses a [very lightweight static server](https://gitlab.com/matipan/static-server) that serves the content and provides a healtcheck that follows the standards from OpenFaaS.
```

Comment:
Why didn't the Nginx template work?

https://github.com/alexellis/static-site-template

Reply:
(It was used in the previous blog, also)

Reply:
The nginx template only serves the static content found in the `function` directory. This template however takes care of building the website for you, this way users can let OpenFaaS Cloud take care of generating the static content.

Reply:
I didn't explain what I meant very well.

Why don't you use the Nginx server to serve the site, and the Hugo / alpine base-image to build it?

Reply:
https://github.com/matipan/openfaas-hugo-template/blob/master/template/hugo/Dockerfile#L20

Reply:
I am not sure how much RAM the unprivileged version of Nginx needs.

Reply:
Oh, it seemed a bit much to have nginx in there for something this simple. Having `Nginx->Gateway->Nginx` made a bit of noise in my head so I didn't though much about using it. Do you see any added benefit for going with nginx?